### DISCUSSION TITLE
How to determine what information is available to resource hooks?

### DISCUSSION THREAD

**Discussion by cwstrommer** [argoproj/argo-cd#7387] [Q&A]
I'm looking to do some operations on Helm charts managed by argocd.  The steps are not argocd-native because they involve fetching information from third party services, and as such would require custom commands to run.  I was thinking that perhaps I could use a pre-sync resource hook to do this, but the documentation on the hooks is rather sparse.  

* What information (if any) about the release that argocd intends to sync is being passed to the hook?
* If no information is passed, are there other options that I could explore?  What I need is:
  * Ability to run arbitrary commands in an arbitrary container
  * Have information about the current sync operation available to the container, such as the chart name, version, and repository or, failing that, the fetched archive (tarballed or not) of the chart that is being considered for sync.

**Comment by larsroger** [comment]
I also would like an answer to this.
